crates/btcio/src/status.rs

```diff
@@ -15,9 +12,9 @@ pub enum L1StatusUpdate {
     IncrementInscriptionCount,
 }
 
-pub async fn apply_status_updates(status_updates: &[L1StatusUpdate], status_tx: Arc<StatusTx>) {
-    let mut l1_status = status_tx.l1.borrow().clone();
-    for event in status_updates {
+pub async fn apply_status_updates(st_updates: &[L1StatusUpdate], st_chan: &StatusChannel) {
```

Comment:
We can simplify this more.  The way this `L1StatusUpdate` concept works is just adding more unnecessary steps.